File: app/src/main/java/by/anegin/tgcontest/core/di/AppComponent.java
package by.anegin.tgcontest.core.di;

import by.anegin.tgcontest.data.DataRepository;
import by.anegin.tgcontest.core.utils.ThemeHelper;

public interface AppComponent {

    DataRepository getDataRepository();

    ThemeHelper getThemeHelper();

}

File: app/src/main/java/by/anegin/telegram_contests/core/di/AppComponent.kt
package by.anegin.telegram_contests.core.di

import by.anegin.telegram_contests.core.data.DataRepository
import by.anegin.telegram_contests.core.data.source.DataSource

interface AppComponent {

    val dataSource: DataSource

    val dataRepository: DataRepository

    companion object {
        lateinit var instance: AppComponent
    }

}

fun app() = AppComponent.instance